storage/src/main/java/zipkin2/storage/kafka/KafkaStoreHttpService.java

```diff
@@ -0,0 +1,346 @@
+import org.apache.kafka.streams.state.ReadOnlyWindowStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import zipkin2.DependencyLink;
+import zipkin2.Span;
+import zipkin2.codec.DependencyLinkBytesEncoder;
+import zipkin2.codec.SpanBytesEncoder;
+import zipkin2.internal.DependencyLinker;
+import zipkin2.storage.QueryRequest;
+
+import static zipkin2.storage.kafka.streams.DependencyStoreTopologySupplier.DEPENDENCIES_STORE_NAME;
+import static zipkin2.storage.kafka.streams.TraceStoreTopologySupplier.AUTOCOMPLETE_TAGS_STORE_NAME;
+import static zipkin2.storage.kafka.streams.TraceStoreTopologySupplier.REMOTE_SERVICE_NAMES_STORE_NAME;
+import static zipkin2.storage.kafka.streams.TraceStoreTopologySupplier.SERVICE_NAMES_STORE_NAME;
+import static zipkin2.storage.kafka.streams.TraceStoreTopologySupplier.SPAN_IDS_BY_TS_STORE_NAME;
+import static zipkin2.storage.kafka.streams.TraceStoreTopologySupplier.SPAN_NAMES_STORE_NAME;
+import static zipkin2.storage.kafka.streams.TraceStoreTopologySupplier.TRACES_STORE_NAME;
+
+/**
+ * Server to enable access to local stores.
+ * <p>
+ * Given the partitioned nature of local stores, a RPC layer is required to allow accessing
+ * distributed state. This component exposes access to local state via Http call from {@link
+ * KafkaSpanStore}
+ *
+ * @since 0.6.0
+ */
+public class KafkaStoreHttpService {
+  static final Logger LOG = LoggerFactory.getLogger(KafkaStoreHttpService.class);
+  static final ObjectMapper MAPPER = new ObjectMapper();
+  static final String EMPTY_ARRAY = "[]";
+
+  final KafkaStorage storage;
+  final long minTracesStored;
+
+  KafkaStoreHttpService(KafkaStorage storage) {
+    this.storage = storage;
+
+    this.minTracesStored = storage.minTracesStored;
+  }
+
+  @Get("/dependencies")
+  public HttpResponse getDependencies(
+      @Param("endTs") long endTs,
+      @Param("lookback") long lookback) {
+    try {
+      ReadOnlyWindowStore<Long, DependencyLink> dependenciesStore =
+          storage.getDependencyStoreStream().store(DEPENDENCIES_STORE_NAME,
+              QueryableStoreTypes.windowStore());
+      List<DependencyLink> links = new ArrayList<>();
+      Instant from = Instant.ofEpochMilli(endTs - lookback);
+      Instant to = Instant.ofEpochMilli(endTs);
+      dependenciesStore.fetchAll(from, to)
+          .forEachRemaining(keyValue -> links.add(keyValue.value));
+      List<DependencyLink> mergedLinks = DependencyLinker.merge(links);
+      LOG.debug("Dependencies found from={}-to={}: {}", from, to, mergedLinks.size());
+      return HttpResponse.of(
+          HttpStatus.OK,
+          MediaType.JSON,
+          DependencyLinkBytesEncoder.JSON_V1.encodeList(mergedLinks));
+    } catch (InvalidStateStoreException e) {
+      LOG.warn("State store is not ready", e);
+      return HttpResponse.of(
+          HttpStatus.OK,
+          MediaType.JSON,
+          DependencyLinkBytesEncoder.JSON_V1.encodeList(new ArrayList<>()));
+    }
+  }
+
+  @Get("/serviceNames")
```

Comment:
While I was reviewing everything changed to an annotated service! Let's change the return type to `AggregatedHttpResponse` for all of these so the blocking works ok for them.